pkg/rp/util/registry.go

```diff
@@ -133,3 +135,12 @@ func parsePath(path string) (repository string, tag string, err error) {
 	tag = reference.Tag()
 	return
 }
+
+func GetRegistrySecrets(definition recipes.Configuration, templatePath string, secrets map[string]recipes.SecretData) (recipes.SecretData, error) {
+	parsedURL, err := url.Parse("https://" + templatePath)
+	if err != nil {
+		return recipes.SecretData{}, err
+	}
+
+	return secrets[definition.RecipeConfig.Bicep.Authentication[parsedURL.Host].Secret], nil
```

Comment:
feels like there could be nil pointer exception(s) 🤣

Reply:
its empty struct and not a pointer, so it wont hit a nil pointer. I already have nil pointer check test for this

Reply:
this is an independent function, so we would need a check here:  
if secret, ok := definition.RecipeConfig.Bicep.Authentication[parsedURL.Host], ok {..}
Pls also include returning an error for this as a test case in Test_GetRegistrySecrets()